applications/StructuralMechanicsApplication/custom_elements/small_displacement_bbar.h

```diff
@@ -0,0 +1,335 @@
+
+        /// Destructor.
+        ~SmallDisplacementBbar() override;
+
+        ///@}
+        ///@name Operators
+        ///@{
+        ///@}
+        ///@name Operations
+        ///@{
+        /**
+         * Returns the currently selected integration method
+         * @return current integration method selected
+         */
+        //TODO: ADD THE OTHER CREATE FUNCTION
+        Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override;
+
+    /**
+     * @brief This function provides the place to perform checks on the completeness of the input.
+     * @details It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input
+     * or that no common error is found.
+     * @param rCurrentProcessInfo The current process info instance
+     */
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+        /**
+         * Calculate a Matrix Variable on the Element Constitutive Law
+         * @param rVariable: The variable we want to get
+         * @param rOutput: The values obtained int the integration points
+         * @param rCurrentProcessInfo: the current process info instance
+         */
+        void CalculateOnIntegrationPoints(
+                const Variable<Matrix>& rVariable,
+                std::vector<Matrix>& rOutput,
+                const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+        /**
+         * Calculate a Vector Variable on the Element Constitutive Law
+         * @param rVariable: The variable we want to get
+         * @param rOutput: The values obtained int the integration points
+         * @param rCurrentProcessInfo: the current process info instance
+         */
+        void CalculateOnIntegrationPoints(
+                const Variable<Vector>& rVariable,
+                std::vector<Vector>& rOutput,
+                const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+        /**
+         * Calculate a double Variable on the Element Constitutive Law
+         * @param rVariable: The variable we want to get
+         * @param rOutput: The values obtained int the integration points
+         * @param rCurrentProcessInfo: the current process info instance
+         */
+        void CalculateOnIntegrationPoints(
+                const Variable<double>& rVariable,
+                std::vector<double>& rOutput,
+                const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+        /**
+         * Called at the end of eahc solution step
+         * @param rCurrentProcessInfo: the current process info instance
+         */
+        void FinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo) override;
+
+    protected:
+
+        /**
+         * Internal variables used in the kinematic calculations
+         */
+        struct KinematicVariables
```

Comment:
If the kinematic variables are almost the same as the ase class you can derive them 

struct KinematicVariables : public BaseSolidElement::KinematicVariables

and add just what you need, this way the quantity of code is reduced